controllers/helpers.go

```diff
@@ -744,3 +747,19 @@ func GetGPUsForPE(ctx context.Context, client *nutanixClientV3.Client, peUUID st
 	}
 	return gpus, nil
 }
+
+// GetFailureDomain gets the failure domain with a given name from a NutanixCluster object.
+func GetFailureDomain(failureDomainName string, nutanixCluster *infrav1.NutanixCluster) (*infrav1.NutanixFailureDomain, error) {
+	if failureDomainName == "" {
+		return nil, fmt.Errorf("failure domain name must be set when searching for failure domains on a Nutanix cluster object")
+	}
+	if nutanixCluster == nil {
+		return nil, fmt.Errorf("nutanixCluster cannot be nil when searching for failure domains")
+	}
+	for _, fd := range nutanixCluster.Spec.FailureDomains {
+		if fd.Name == failureDomainName {
```

Comment:
your example in description says <pre>
  NUTANIX_FAILURE_DOMAIN_1_PRISM_ELEMENT_NAME: "<pe-1>"
  NUTANIX_FAILURE_DOMAIN_1_SUBNET_NAME: "<subnet-1>"
  NUTANIX_FAILURE_DOMAIN_2_PRISM_ELEMENT_NAME: "<pe-2>"
  NUTANIX_FAILURE_DOMAIN_2_SUBNET_NAME: "<subnet-2>"
  NUTANIX_FAILURE_DOMAIN_3_PRISM_ELEMENT_NAME: "<pe-1>"
  NUTANIX_FAILURE_DOMAIN_3_SUBNET_NAME: "<subnet-2>"
</pre>
so if tehre are 2 fd with same cluster name then how will that work

Reply:
could you please also update the description which ties cluster name from nutanixmachinetemplate to fd name in nutnaixcluster